workflow/ignite/handlers/add_train_progress_bar.py

```diff
@@ -0,0 +1,45 @@
+import ignite
+from ignite.engine import Events
+from tqdm import tqdm
+
+from workflow.ignite.constants import TQDM_OUTFILE
+
+
+def add_train_progress_bar(
+    trainer, config, output_transform=lambda batch: batch['loss']
+):
+
+    ignite.metrics.RunningAverage(
+        output_transform=output_transform, alpha=0.98
+    ).attach(trainer, 'running_average_loss')
+
+    epoch = 1
+
+    def create_progress_bar():
+        return tqdm(
+            total=config['n_batches_per_epoch'],
+            file=TQDM_OUTFILE,
+            desc=f'''Epoch [{epoch} / {config['max_epochs']}]''',
+            leave=False
+        )
+
+    epoch_progress_bar = None
+
+    @trainer.on(Events.EPOCH_STARTED)
+    def reset_epoch_progress_bar(engine):
+        nonlocal epoch_progress_bar
```

Comment:
kinda ugly to have a reference like this, aja, I'll let it slide and think about how to fix it properly
feels like this is going to have to be done in much the same way everytime you need to have state between iterations

Reply:
I think we should consider doing it like the do in ignite.contrib instead.

See ignite.contrib.handlers.tqdm_logger.ProgressBar

I suggest we make a PR fixing this soon. Since it is very fragile, I would prefer we merge this before trying to improve the solution more now